### PR TITLE
Add PodControlInterface and PodControllerRefManager

### PR DESCRIPTION
**Pull request by Jeffwan**
This PR aims to solve following issue and it also help to remove more direct dependency on Kubernetes.  Resolve part of #48 #71 

### Background
Current implementation uses `PodControlInterface` and `PodControllerRefManager` from `k8s.io/kubernetes/pkg/controller` and self-implemented `ServiceControlInterface` and `ServiceControllerRefManager` which bring some problems.

1. PodControlInterface/PodControllerRefManager may change along with k8s version upgrade, while we don't make changes to Service. It maybe out of sync.
2. Half and half doesn't help us remove Kubernetes direct dependency.
3. It’s not that straightforward for maintainers and contributors. The first time I look at the change, I notice we only have `service_ref_manager` which is kind of unclear until I find #36 

### Solution
Consider the situation that Kubernetes upstream doesn't have implementation of `ServiceControlInterface` and `ServiceControllerRefManager`, I would suggest we just folk `PodControlInterface` and `PodControllerRefManager` locally.

This help remove dependency a lot and we can also make sure pods/service implementation can be updated at the same time. 

In addition, I will make some efforts to this issue. https://github.com/kubernetes/client-go/issues/332. I will try to submit a PR and see feedbacks. If change can be accepted, then we don't need to maintain these auxiliary codes on our side. 

/cc @gaocegege @terrytangyuan 

Need @gaocegege 's review on code folk since he originally worked on this.

### CODE REVIEW

File: pkg/controller.v1/control/controller_ref_manager.go [hunk 1]
```diff
@@ -0,0 +1,377 @@
+// Copyright 2019 The Kubeflow Authors
```
Comment (Jeffwan) [inline pkg/controller.v1/control/controller_ref_manager.go:1]: If we think it's fine to sync pod_ref_manager, it makes sense to rename to `controller_ref_manager.go`.

File: pkg/controller.v1/control/service_control.go [hunk 2]
```diff
@@ -16,6 +16,7 @@ package control
 
 import (
 	"fmt"
+	"k8s.io/apimachinery/pkg/api/errors"
```
Comment (gaocegege) [inline pkg/controller.v1/control/service_control.go:19]: Please move this to the next group

Comment (Jeffwan) [reply pkg/controller.v1/control/service_control.go]: Yeah, I move to following block.

**Comment by Jeffwan** [comment]
@gaocegege Address the feedbacks, please have another look

**Comment by gaocegege** [comment]
> LGTM but could you add some notes in the forked code so we can always look back what we did and where the code was forked from (with version/link).

+1 for this suggestion.

**Comment by Jeffwan** [comment]
@gaocegege @terrytangyuan 

Thanks for the review. That's great suggestion. I will add a follow up CONTRIBUTING.md doc and include these technical details.

**Comment by gaocegege** [comment]
OK, then I think we can merge this one.

**Pull request merged**